sphinx/builders/__init__.py

```diff
@@ -253,15 +253,13 @@ def compile_all_catalogs(self):
         message = 'all of %d po files' % len(catalogs)
         self.compile_catalogs(catalogs, message)
 
-    def compile_specific_catalogs(self, specified_files):
+    def compile_specific_catalogs(self, specified_docnames):
         # type: (List[unicode]) -> None
-        def to_domain(fpath):
+        def to_domain(docname):
             # type: (unicode) -> unicode
-            docname, _ = path.splitext(path_stabilize(fpath))
```

Comment:
Nope, if I understand you correctly, then I have fixed this in https://github.com/sphinx-doc/sphinx/pull/4560/files#diff-bb8897b780588faa431faef0abc85e97R322

Reply:
I meant this is also broken in stable release. I understand this PR works fine.

Reply:
Then yes, it is broken. Didn't we discuss that on #4476. I haven't bothered targeting this to stable because it's a latent bug with very limited impact. I can change that, however, if you think otherwise.

Reply:
Sorry, I'd been confused this as refactoring PR. Okay, I understand now.

Reply:
Nope, I just had to move a lot of stuff to get this to work, and then took the opportunity to clean up other stuff while I was at it

Reply:
@tk0miya I've gone and removed a lot of the unrelated stuff from this and retargeted it to `1.7-release`. I will submit the other patches separately.

Reply:
...or not. You already fixed this in #4571 😞